From 007777696321407ffd8924fe3e6f6f4103e0f20a Mon Sep 17 00:00:00 2001
From: Angus Gratton <gus@projectgus.com>
Date: Sat, 5 Sep 2015 12:42:30 +1000
Subject: [PATCH 05/22] Add xtensa/machine/_default_types.h overriding generic

Defines uint32_t/int32_t as int not long, even if gcc's __INT32_TYPE__ is long.

Signed-off-by: Angus Gratton <gus@projectgus.com>
---
 .../sys/xtensa/include/machine/_default_types.h    | 52 ++++++++++++++++++++++
 1 file changed, 52 insertions(+)
 create mode 100644 newlib/libc/sys/xtensa/include/machine/_default_types.h

diff --git a/newlib/libc/sys/xtensa/include/machine/_default_types.h b/newlib/libc/sys/xtensa/include/machine/_default_types.h
new file mode 100644
index 000000000..1d5569786
--- /dev/null
+++ b/newlib/libc/sys/xtensa/include/machine/_default_types.h
@@ -0,0 +1,52 @@
+/*
+ * _default_types implementation for xtensa lx106 arch
+ *
+ * Simplified version of generic _default_types.h, ignores gcc
+ * built-in standard types.
+ */
+
+#ifndef _MACHINE__DEFAULT_TYPES_H
+#define _MACHINE__DEFAULT_TYPES_H
+
+#include <sys/features.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef signed char __int8_t ;
+typedef unsigned char __uint8_t ;
+#define ___int8_t_defined 1
+
+typedef signed short __int16_t;
+typedef unsigned short __uint16_t;
+#define ___int16_t_defined 1
+
+typedef signed int __int32_t;
+typedef unsigned int __uint32_t;
+#define ___int32_t_defined 1
+
+typedef signed long long __int64_t;
+typedef unsigned long long __uint64_t;
+#define ___int64_t_defined 1
+
+typedef __int8_t __int_least8_t;
+typedef __uint8_t __uint_least8_t;
+
+typedef __int16_t __int_least16_t;
+typedef __uint16_t __uint_least16_t;
+
+typedef __int32_t __int_least32_t;
+typedef __uint32_t __uint_least32_t;
+
+typedef __int64_t __int_least64_t;
+typedef __uint64_t __uint_least64_t;
+
+typedef __INTPTR_TYPE__ __intptr_t;
+typedef __UINTPTR_TYPE__ __uintptr_t;
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* _MACHINE__DEFAULT_TYPES_H */
-- 
2.11.0 (Apple Git-81)

